src/components/Chart/Chart.tsx

```diff
@@ -66,7 +68,10 @@ class Chart extends React.Component<
   }
 
   componentDidUpdate(prevProps: any): void {
-    if (!prevProps.instrument) {
+    if (
+      !prevProps.instrument ||
```

Comment:
just figured out it would be a good case for `shouldComponentUpdate` but not sure it will not break the rest parts